.github/workflows/tics.yaml

```diff
@@ -11,6 +11,13 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+
+      - name: Setup Python
```

Comment:
Is this for consistency or is it failing with the system python3.10?

Reply:
I copied the shallow checkout to potentially fix https://github.com/canonical/snapcraft/actions/runs/9844524213/job/27178103576 which is a version parsing issue (my view is that if configured the same it should work the same)

The python thing is to keep it consistent with the pull_request triggered flow, by keeping them in sync it is much easier to anticipate errors as it is not easy to test something that runs on main pushes only